decidim-core/app/controllers/decidim/features/base_controller.rb

```diff
@@ -52,6 +53,10 @@ def ability_context
           feature_settings: feature_settings
         )
       end
+
+      def redirect_unless_feature_private
+        redirect_to "/404" unless current_user_can_visit_space?
```

Comment:
Same, does this return a 404 status?

Reply:
Now we make a redirect to the index view and show a flash alert telling that are not allowed to view the content. it's going well?

Reply:
Where should I redirect this route? How can we redirect from within a feature to the participatory space index?

Reply:
If you raise an exception, rails will do that for you:

```ruby
raise ActionController::RoutingError.new('Not Found')
```

This will show the error on development, but render a 404 on production.